src/metabase/public_settings.clj

```diff
@@ -638,3 +638,9 @@
   (deferred-tru "Prefix for upload table names")
   :visibility   :authenticated
   :type         :string)
+
+(defsetting user-visibility
+  (deferred-tru "Determines what other users non-admin users are able to see. Possible values are 'all users', 'same group', and 'none'.")
+  :visibility   :authenticated
+  :type         :string
```

Comment:
I'd prefer this being a keyword setting with values like `:all`, `:group` and `:none`, instead of using a string. Keywords are just a bit more idiomatic in clojure and they'll be coerced to strings when the FE fetches them anyway